src/Sylius/Behat/Context/Domain/PromotionContext.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Promotion\Model\CouponInterface;
+use Sylius\Component\Promotion\Repository\PromotionRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class PromotionContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var PromotionRepositoryInterface
+     */
+    private $promotionRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $couponRepository;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param PromotionRepositoryInterface $promotionRepository
+     * @param RepositoryInterface $couponRepository
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        PromotionRepositoryInterface $promotionRepository,
+        RepositoryInterface $couponRepository
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->promotionRepository = $promotionRepository;
+        $this->couponRepository = $couponRepository;
+    }
+
+    /**
+     * @When /^I try to delete ("([^"]+)" coupon)$/
+     */
+    public function iTryToDeleteCoupon(CouponInterface $coupon)
+    {
+        try {
+            $this->couponRepository->remove($coupon);
+
+            throw new \Exception(sprintf('Coupon "%s" was removed which should not happen.', $coupon->getCode()));
```

Comment:
``'Coupon "%s" has been remove, but it shouldn't've.``? :smile: /cc @TheMadeleine